### ISSUE TITLE
No exclusion pattern works

### ISSUE THREAD

**Issue by shayaniox** [foxyproxy/firefox-extension#223]
FoxyProxy version: 8.9
Firefox version: 128.0 Ubuntu
---
![2024-08-04-233404_1141x100_scrot](https://github.com/user-attachments/assets/ed038dde-b8b3-4890-ad03-fd257a05592a)
As it's shown in the picture, I've defined two patterns to exclude a bunch of URLs.
Also I have tested my patterns using the `pattern tester`:
![2024-08-04-233658_1220x254_scrot](https://github.com/user-attachments/assets/152a0182-8483-493c-985a-da574ac530e4)
When try to access the URL: `http://localhost:8025` the browser tabs goes loading and eventually can't access the intended address:
![2024-08-04-233934_916x430_scrot](https://github.com/user-attachments/assets/466b782f-551a-4626-9986-65a51488a3c6)

Please let me know if there is any further information required.

**Comment by erosman** [comment]
Please use the [FoxyProxy 8+](https://github.com/foxyproxy/browser-extension) for v8+.

Are you using "Proxy by Patterns" or an individual proxy for all connections?

Include/Exclude patterns are meant to be used with  "Proxy by Patterns". They will be ignored if a single proxy is used for everything.

In case you are using a single proxy for everything and want to exclude some hosts, it would be best to use the Global Exclude.

See also: https://github.com/foxyproxy/browser-extension/issues/140

**Comment by shayaniox** [comment]
@erosman Thanks for the rapid answer.
When I define just exclusion patterns, the options `Proxy by pattern` doesn't show up, but once an inclusion patterns, it does.
Does `Global Exclude` accept wildcard or regex patterns?
Is it possible for a specific defined proxy to set exclusion patterns and the patterns are activated only for that proxy?

It's worth to mention the proxy type is SOCKS4.

**Comment by shayaniox** [comment]
I just tried `Global Exclude` but the issue with the localhost persists.
![2024-08-05-133721_1221x823_scrot](https://github.com/user-attachments/assets/85e63f88-b1e6-4bee-8d03-699f81c3f5fa)

**Comment by erosman** [comment]
> When I define just exclusion patterns, the options Proxy by pattern doesn't show up, but once an inclusion patterns, it does.

"Proxy by Patterns" works by matching the INCLUDE (minus EXCLUDE). If there are no include patterns, it does not show as an option, since it wont be able to match anything.

> Does Global Exclude accept wildcard or regex patterns?

Global exclude uses the pattern defined by the browser (see the included Help under Global Exclude).
Also: [Global Exclude](https://foxyproxy.github.io/browser-extension/src/content/help.html#global-exclude)

> Is it possible for a specific defined proxy to set exclusion patterns and the patterns are activated only for that proxy?

Include/Exclude patterns are only processed with "Proxy by Patterns". If you only want to use a single proxy with include/exclude patterns, you can disable other proxies or not set any include/exclude for them and then use "Proxy by Patterns".

> I just tried Global Exclude but the issue with the localhost persists.

Try `<local>` in Global Exclude.

**Comment by shayaniox** [comment]
@erosman 

> Try <local> in Global Exclude.

Didn't work for either `localhost:8025` or `127.0.0.1:15672`.

It feels like a hack by the method that worked is like:
1. Add the pattern `*` to inclusion patterns
2. Add desired exclusion patterns
3. Turn on the `Proxy by Patterns` option

**Comment by shayaniox** [comment]
@erosman 
Isn't it a bug that the option `Proxy by Patterns` is not listed for the extension, until you enter the first inclusion pattern?
Shouldn't it be available just by setting exclusion patterns?

**Comment by erosman** [comment]
Did you read https://github.com/foxyproxy/firefox-extension/issues/223#issuecomment-2268877799?

**Comment by shayaniox** [comment]
> Did you read [#223 (comment)](https://github.com/foxyproxy/firefox-extension/issues/223#issuecomment-2268877799)?

Yes, but you wrote `it does show`, but it doesn't. I think it's a typo and `it does **NOT** show` is correct. I get why it needs `inclusion patterns`.
What about the Global Patterns?
Any idea why it doesn't work?

**Comment by erosman** [comment]
Ah... yes.. there was a typo.. corrected.

> What about the Global Patterns?

Can you post your exported preferences  (remove private data) for me to see?
